### ISSUE TITLE
Sidebar Game Log Improvements

### ISSUE THREAD

**Issue by alexzhaohong** [bobtheuberfish/chiriboga#121]
<img width="356" alt="Screenshot 2023-03-07 at 5 13 13 PM" src="https://user-images.githubusercontent.com/12620082/223593752-5510f12e-6d85-414d-bf2c-8b555956fe04.png">

My main issue: the Blue Text on the black background is a bit hard to see. Do you know of a good alternative? Perhaps I could look into High Contrast A11y standards to find a text color, if you would like.

Second nit: The runner AI plays a bit too quickly, especially for a newer player. Will there be a "replay" option where the game actions are animated at a slower pace? or perhaps a 2 second wait could be implemented between actions. 

Third nit: For the sidebar itself, I feel like it is a bit too tiny for my eyes. The current iconography is good though, I like the image + text combo. Perhaps more recent logs could be larger in size?

**Comment by bobtheuberfish** [comment]
Thank you for these suggestions. It will be easy to change the blue text colour.

The current minimum wait time between AI actions is 0.35 seconds. I can see that this is fast for newer players, but 2 seconds would be too slow for experienced players. At the moment you can slow the AI actions down by turning on narration in the options - AI will only take the next action when narration of its first action is complete. Do you think I should add an option to change the AI play speed for when not using narration?
I'm not sure what you mean by a "replay" option. Can you explain this more?

I too am not fully satisfied with the sidebar but I have not thought of a good solution yet. At the moment it is super tiny on mobile screens and I'm not sure what to do about that. If I make the sidebar bigger, it might use up playing field space and require the game to zoom out more so that the cards are smaller, which isn't great either.
Can you explain more about how the more recent logs could be smaller in size? You could draw a quick diagram perhaps?

**Comment by alexzhaohong** [comment]
Regarding the speed of AI turns, I have been able to test out the narration. I think it makes the AI turn grokkable! Since Chiriboga does not have any sfx, perhaps narration could be turned on by default, especially for the tutorial games?

I haven't given more thought to the sidebar specifically, but I see you have improved the contrast for the Corp color by lightening the blue. Nice! Do you see how the first Sidebar Bubble is larger than the rest? I was thinking that all bubbles up to and including the last AI turn could be that large, and everything after that could be smaller. I also think the Scrollbar normal/hover colors should be swapped.

The current "Rewind" functionality let's me rewatch the AI's turn, which is the main reason why I wanted a "Replay" button

**Comment by bobtheuberfish** [comment]
Rather than changing the defaults I have created two new options for the in-game menu. Each will take effect after the next player action. These are currently live in the dev version of the game but not the main version.
https://chiriboga.sifnt.net.au/dev/

Please try this and let me know if the slower AI is still too fast (it currently runs 3x slower than the normal AI).

I was not able to change the scrollbar hover colour. The CSS code only takes two values: a track colour and a button colour. The hover colour is determined by the browser.

**Comment by bobtheuberfish** [comment]
I thought you would like to know that I am joining another in-progress Netrunner project. I will still maintain Chiriboga in terms of handling issues and fixing bugs but I don't intend to add new cards. This other project does not have AI yet but it has many many cards implemented and the lead developer is someone with professional experience so it has better long-term potential. If you are interested in being involved I'm sure your contributions would be appreciated, just let me know.

**Comment by Morgan-6Freedom** [comment]
> I thought you would like to know that I am joining another in-progress Netrunner project. I will still maintain Chiriboga in terms of handling issues and fixing bugs but I don't intend to add new cards. This other project does not have AI yet but it has many many cards implemented and the lead developer is someone with professional experience so it has better long-term potential. If you are interested in being involved I'm sure your contributions would be appreciated, just let me know.

Is there any discord / forum / website / github about this project ? Is it still web based ? Can we help ?

**Comment by bobtheuberfish** [comment]
Help is welcome, please PM me on Discord, you can find me via Green Level Clearance. It is not web-based (code is C#) but can be built for various platforms including browser in theory.

**Comment by bobtheuberfish** [comment]
The actual issue this thread was originally about has been inactive for a few months so I am going to close it. I will create a new issue specifically about the new project for anyone else who is interested.
